python/tskit/trees.py

```diff
@@ -5442,3 +5442,44 @@ def newick_trees(self, precision=3, breakpoints=None, Ne=1):
             "This method is no longer supported. Please use the Tree.newick"
             " method instead"
         )
+
+    def nexus(self, precision=14):
+        """
+        Returns a nexus encoding <https://en.wikipedia.org/wiki/Nexus_file>`_
+        of this tree. Trees along the sequence are listed sequentially in
+        the TREES block. The tree spanning the interval :math:`[x, y)`` is
+        given the name "tree_x_y". Spatial positions are written at the
+        specified precision.
+
+        :param int precision: The numerical precision with which branch lengths
+            and tree positions are printed.
+        :return: A nexus representation of this tree.
+        :rtype: str
+        """
+        leaf_labels = {}
+        for i, u in enumerate(self.samples()):
+            leaf_labels[u] = str(i + 1)
+
+        s = "#NEXUS\n"
+        s += "BEGIN TAXA;\n"
+        s += "\tTAXLABELS\n"
+
+        for leaf_id in leaf_labels.values():
```

Comment:
Your depending on the order of iteration here, which isn't guaranteed in older versions of Python. Seems clearer to be explicit about the order that your assuming here by iterating over samples:
```
for u in self.samples():
     s += f"\t\t{leaf_labels[u]}\n"
```